sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/module-info.java

```diff
@@ -0,0 +1,51 @@
+module com.azure.resourcemanager.resources {
+    requires transitive com.azure.core.management;
+    requires com.github.spotbugs.annotations;
+
+    exports com.azure.resourcemanager.resources.fluentcore;
+    exports com.azure.resourcemanager.resources.fluentcore.arm;
+    exports com.azure.resourcemanager.resources.fluentcore.arm.collection;
+    exports com.azure.resourcemanager.resources.fluentcore.arm.models;
+    exports com.azure.resourcemanager.resources.fluentcore.collection;
+    exports com.azure.resourcemanager.resources.fluentcore.dag;
+    exports com.azure.resourcemanager.resources.fluentcore.model;
+    exports com.azure.resourcemanager.resources.fluentcore.rest;
+    exports com.azure.resourcemanager.resources.fluentcore.utils;
+
+    exports com.azure.resourcemanager.resources;
+    exports com.azure.resourcemanager.resources.models;
+    exports com.azure.resourcemanager.resources.fluent;
+    exports com.azure.resourcemanager.resources.fluent.inner;
+
+    opens com.azure.resourcemanager.resources.fluent.inner to com.fasterxml.jackson.databind, com.azure.core;
+    opens com.azure.resourcemanager.resources.models to com.fasterxml.jackson.databind, com.azure.core;
+
+    exports com.azure.resourcemanager.resources.fluentcore.arm.collection.implementation
```

Comment:
Exporting `implementation` package is not recommended. If there are some common classes that are shared across multiple modules, then those classes should be in non-implementation package and that package should be exported.

Reply:
Do you mean we can rename the package and/or the classes? An example is the `GroupableResourceImpl` used in  [VirtualMachineImpl](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineImpl.java#L106). There are a few `{CommonResourceModel}Impl` shared from `azure-resourcemanager-resources` to other `azure-resourcemanager-**` services.

Reply:
The intent of the classes inside the 'implementation' packages is that they are never intended for use by end users. In JDK 9+ with modules, we can hide everything under this package and therefore have stronger protections against these being used. If we export them, we're undoing that work. We should have clearly defined barriers between public API and private API - nothing in the public API should expose private API to end users, and so therefore we should never need the user to directly interact with anything in the implementation package. If this is the case today, consider refactoring to get the classes in the correct package and with the correct name.

Reply:
@srnagar @JonathanGiles 
Would you think if `exports implementation to moduleA, moduleB` could be acceptable as the end users cannot get them. Alternatively, we may change namespace from `com.azure.resourcemanager.resources.fluentcore.arm.models.implementation` to `com.azure.resourcemanager.resources.fluentcore.arm.models.internal`.

Reply:
It is best to think of doing 'exports to' as a code smell that should be avoided unless absolutely necessary. It would be preferable to clearly determine firstly if the code is something that should be considered public API or not. That will help to inform how to proceed. It sounds like it should be considered public API to me.

Reply:
Agree what you said about `The intent of the classes inside the 'implementation' packages is that they are never intended for use by end users.` And this is what `export to` could help for a qualified export. With that, only the designated modules can get access to the 'implementation' packages while end users cannot. It's kind of making them internal available only inside resource manager modules. This is a little bit different comparing to data plane SDK as for resource management these 'implementation' classes are helping model ARM resource in general manner. We will reuse them for the resources among various RPs.